velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -117,15 +208,52 @@ void HiveDataSink::createWriter(
   options.schema = inputType_;
   // Without explicitly setting flush policy, the default memory based flush
   // policy is used.
-  auto writerParameters = getWriterParameters(partitionName);
-
-  writerInfo_.emplace_back(std::make_shared<HiveWriterInfo>(*writerParameters));
 
+  std::optional<std::string> partitionName = partitionChannels_.size() > 0
+      ? std::optional<std::string>(makePartitionName(
+            input_,
+            partitionChannels_,
+            partitionRows_[partitionId]->as<vector_size_t>()[0]))
+      : std::nullopt;
+  auto writerParameters = getWriterParameters(partitionName);
   auto writePath = fs::path(writerParameters->writeDirectory()) /
       writerParameters->writeFileName();
+
   auto sink = dwio::common::DataSink::create(writePath);
-  writers_.push_back(std::make_unique<Writer>(
-      options, std::move(sink), *connectorQueryCtx_->memoryPool()));
+  writers_[partitionId] = std::make_unique<Writer>(
+      options, std::move(sink), *connectorQueryCtx_->memoryPool());
+  writerInfo_[partitionId] =
+      std::make_shared<HiveWriterInfo>(*writerParameters);
+}
+
+void HiveDataSink::computePartitionRowCountsAndIndices(
+    uint64_t numPartitionIds,
+    memory::MemoryPool* pool) {
+  partitionSizes_.resize(numPartitionIds);
+  std::fill(partitionSizes_.begin(), partitionSizes_.end(), 0);
+  partitionRows_.resize(numPartitionIds);
+  std::fill(partitionRows_.begin(), partitionRows_.end(), nullptr);
```

Comment:
This code prevents the reuse of the buffers. Please, revise.

Reply:
Revised. The individual Buffers of partitionRows_ need to resize to have more capacity if needed. What functions shall I call to increase capacity?

I'm using
```c++
partitionRows_[id] = allocateIndices(numRows, pool);
```
now to reallocate the memory. Do I need to call some function to release the memory of the old Buffer?

Reply:
Check out Buffer::reallocate method.